client/my-sites/current-site/index.jsx

```diff
@@ -159,9 +156,13 @@ const CurrentSite = React.createClass( {
 // TODO: make this pure when sites can be retrieved from the Redux state
 module.exports = connect(
 	( state, ownProps ) => {
+		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSite = getSelectedSite( state );
+
 		return {
-			selectedSite: selectedSite || ownProps.sites.getPrimary()
+			selectedSiteId,
```

Comment:
Oh good catch @spen. At worst I think we'll not display a domain notice.

Perhaps something like:
```javascript
const selectedSite =  getSelectedSite( state ) || ownProps.sites.getPrimary();
const selectedSiteId = get( selectedSite, 'ID' );
const isJetpack = get( selectedSite, 'jetpack' );
```

Another alternative would be to set selectedSiteId to primary as a router handler.